modules/core/src/main/java/org/apache/fluo/core/client/FluoAdminImpl.java

```diff
@@ -340,12 +338,20 @@ private String copyJarsToDfs(String jars, String destDir) {
     String dfsAppRoot = config.getDfsRoot() + "/" + config.getApplicationName();
     String dfsDestDir = dfsAppRoot + "/" + destDir;
 
-    FileSystem fs;
+    FileSystem fs = null;
     try {
       fs = FileSystem.get(new URI(config.getDfsRoot()), new Configuration());
       fs.mkdirs(new Path(dfsDestDir));
     } catch (Exception e) {
       throw new IllegalStateException(e);
+    } finally {
+      if (fs != null) {
+        try {
+          fs.close();
```

Comment:
The `fs` variable is used later on in the method so it shouldn't be closed in the `finally` block.  Maybe all uses of fs in the method should be wrapped with a try-with-resources block.